### PR TITLE
ci: add linting for playbooks in the pipeline

### PR DESCRIPTION
**Pull request by Itxaka**
Will trigger an ansible-lint run if we detect changes under the
playbooks dir

### CODE REVIEW

**Comment by Itxaka** [comment]
the `when` should execute correctly as we can see the changes properly in jenkins, it should detect that something has changed there and run the step. We should probably do the same for the docs :)

**Comment by evrardjp** [comment]
> the `when` should execute correctly as we can see the changes properly in jenkins, it should detect that something has changed there and run the step. We should probably do the same for the docs :)

For docs we should be only running the docs jobs if it's a docs only change to also save resources.

File: Jenkinsfile.integration [hunk 1]
```diff
@@ -38,6 +38,18 @@ pipeline {
             }
         }
 
+        stage('Lint') {
```
Comment (evrardjp) [inline Jenkinsfile.integration]: nit: I would say lint playbooks to make it clear it's only linting plays.

File: Jenkinsfile.integration [hunk 2]
```diff
@@ -38,6 +38,18 @@ pipeline {
...
+            options {
+                timeout(time: 5, unit: 'MINUTES', activity: true)
+            }
+            when {
+                changeset "playbooks/**"
```
Comment (toabctl) [inline Jenkinsfile.integration]: maybe also when tox.ini changes?

**Comment by Itxaka** [comment]
> For docs we should be only running the docs jobs if it's a docs only change to also save resources.

Yep, I was trying to imply that, sorry for not making it clearer!

File: Jenkinsfile.integration (hunk 1 above)
Comment (Itxaka) [reply Jenkinsfile.integration]: done

File: Jenkinsfile.integration (hunk 2 above)
Comment (Itxaka) [reply Jenkinsfile.integration]: done

**Comment by toabctl** [comment]
@Itxaka you sould change something in tox.ini or playbooks/ to actually trigger the build step :)

**Comment by Itxaka** [comment]
> @Itxaka you sould change something in tox.ini or playbooks/ to actually trigger the build step :)

Already done :P
https://github.com/SUSE-Cloud/socok8s/pull/445 and https://github.com/SUSE-Cloud/socok8s/pull/446

**Comment by Itxaka** [comment]
did not work :(

**Comment by Itxaka** [comment]
`14:00:50  Warning, empty changelog. Probably because this is the first build.`

¬_¬ so on first build there is no way of checking what has changed? what?

**Comment by Itxaka** [comment]
opened since 2009.... https://issues.jenkins-ci.org/browse/JENKINS-4162

**Comment by Itxaka** [comment]
we need tox in those nodes, not sure where to do that or request that :/

**Comment by toabctl** [comment]
> we need tox in those nodes, not sure where to do that or request that :/

you can ask me :) But it seems to be already installed ....

**Comment by Itxaka** [comment]
> you can ask me :) But it seems to be already installed ....

woops, yeah I installed it yesterday with @jhesketh help :)

And the job went green as well (at least the linting part)

**Pull request merged**